provisioning_templates/snippet/set_hostname.erb

```diff
@@ -0,0 +1,20 @@
+<%#
+name: set_hostname
+snippet: true
+model: ProvisioningTemplate
+kind: snippet
+-%>
+if [ `command -v hostnamectl` ]; then
+  hostnamectl set-hostname <%= @hostname %>
+else
+  hostname <%= @hostname %>
+  # persist the change to survive  reboot
+  echo "<%= @hostname -%>" > /etc/hostname
+fi
+
+if grep '^127.0.1.1\s*<%= @hostname -%>\s*<%= @shortname -%>$' /etc/hosts; then
+  echo "skipping /etc/hosts update, record is already present"
+else
+  echo "adding record to the first line of /etc/hosts"
+  sed -i '1s/^/127.0.1.1   <%= @hostname -%> <%= @shortname -%>\n/' /etc/hosts
```

Comment:
I don't think that's the original motivation (at least in my case). The image already contains /etc/hosts, I choose the FQDN when I do the provisioning, the OS can't know what I had in mind during provisioning. Therefore we to set the hostname as part of provisioning.

The question is what was the motivation for "fixing" the hosts file in fix_hosts.erb snippet. If someone knows or consider that localhost and ipv6 addresses always right, I'm happy to incorporate it here. The question is whether this should be part of "set_hostname" though, maybe it should remain in fix_hosts snippets then.